consolidate.js

```diff
@@ -0,0 +1,36 @@
+var utils = require('./utils')
+
+// similar to accumulative but add all inputs
+module.exports = function consolidate (utxos, outputs, feeRate) {
+  if (!isFinite(utils.uintOrNaN(feeRate))) return {}
+  var bytesAccum = utils.transactionBytes([], outputs)
+
+  var inAccum = 0
+  var inputs = []
+  var outAccum = utils.sumOrNaN(outputs)
+
+  for (var i = 0; i < utxos.length; ++i) {
+    var utxo = utxos[i]
+    var utxoBytes = utils.inputBytes(utxo)
+    var utxoFee = feeRate * utxoBytes
+    var utxoValue = utils.uintOrNaN(utxo.value)
+
+    // skip detrimental input
+    if (utxoFee > utxo.value) {
```

Comment:
This assumes the utxos array is ordered by descending value.

Can this assumption be made?